core/trie2/databasetest.go

```diff
@@ -27,17 +27,30 @@ func newTestNodeReader(id trieutils.TrieID, nodes []*trienode.MergeNodeSet, db d
 }
 
 func (n *testNodeReader) Node(owner *felt.Felt, path trieutils.Path, hash *felt.Felt, isLeaf bool) ([]byte, error) {
+	var (
+		node trienode.TrieNode
+		set  *trienode.NodeSet
+		ok   bool
+	)
 	for _, nodes := range n.nodes {
-		var (
-			node trienode.TrieNode
-			ok   bool
-		)
-		node, ok = nodes.OwnerSet.Nodes[path]
-		if !ok {
-			continue
+		if nodes.OwnerSet.Owner.IsZero() {
+			node, ok = nodes.OwnerSet.Nodes[path]
+			if !ok {
+				continue
+			}
+		} else {
+			set, ok = nodes.ChildSets[*owner]
+			if !ok {
+				continue
```

Comment:
There are no tests for when the Owner is different than zero, why? Is it possible to add one?